README.md

```diff
@@ -0,0 +1,108 @@
+# gmusic-ui.js
+[![Build Status](https://travis-ci.org/gmusic-utils/gmusic-ui.js.svg?branch=master)](https://travis-ci.org/gmusic-utils/gmusic-ui.js)
+[![GitHub release](https://img.shields.io/github/tag/gmusic-utils/gmusic-ui.js.svg)]()
+[![Code Climate](https://img.shields.io/codeclimate/github/gmusic-utils/gmusic-ui.js.svg)]()
+[![GitHub license](https://img.shields.io/github/license/gmusic-utils/gmusic-ui.js.svg)]()
+
+Browser-side JS library for controlling [Google Music][] through UI manipulation.
+
+[Google Music]: https://play.google.com/music/
+
+`gmusic-ui.js` is not created by, affiliated with, or supported by Google Inc.
+
+[Google Music]: https://play.google.com/music/listen
+
+## Getting Started
+It's important to note the `gmusic-ui.js` depends on [`gmusic.js`](https://github.com/gmusic-utils/gmusic.js) in order to operate correctly
+
+### npm
+Install the module with: `npm install gmusic.js gmusic-ui.js`
+
+Once installed, add it to your HTML and access it via `window.GMusic`.
+
+```html
+<script src="node_modules/gmusic.js/dist/gmusic.min.js"></script>
+<script src="node_modules/gmusic-ui.js/dist/gmusic-ui.min.js"></script>
+<script>
+  window.gpm = new window.GMusic(); // Our Google Music API
+</script>
+```
+
+### Vanilla
+If you are not using a package manager, download the latest script at:
+
+https://raw.githubusercontent.com/gmusic-utils/gmusic-ui.js/master/dist/gmusic-ui.min.js
+
+Then, add it to your HTML and access it via `window.GMusic`.
+
+```html
+<script src="gmusic.min.js"></script>
+<script src="gmusic-ui.min.js"></script>
+<script>
+  window.gpm = new window.GMusic(window); // Our Google Music API
+</script>
+```
+
+## Documentation
+`gmusic-ui.js` **extends** the `GMusic` constructor, `window.GMusic`
+
+### `new GMusic(window)`
+Constructor for a new Google Music API.
+
+- window - `Object` -  The global window object
+
+You can read up on the namespaces and methods that `gmusic` natively offers over in the [gmusic.js README](https://github.com/gmusic-utils/gmusic.js/blob/master/README.md#documentation)
+
+Below are the namespaces and methods that `gmusic-ui` **adds** to GMusic
+
+### Playlists
+
+#### `playlists.getAll()`
+Retrieves a list of all the playlists in the users GPM library
+
+**Returns:**
+- retVal `Playlist[]` - An array of [`Playlist`](#playlist) objects
+
+#### `playlists.play(playlist)`
+Navigates to the given playlist and plays it immediately
+
+- playlist `Playlist` - A [`Playlist`](#playlist) object returned from [`getAll()`](#playlistsgetall)
+
+#### `playlists.playWithTrack(playlist, track)`
+Navigates to the given playlist and plays it immediately starting at the given track
+
+- playlist `Playlist` - A [`Playlist`](#playlist) object returned from [`getAll()`](#playlistsgetall)
+- track `Track` - A [`Track`](#track) object from the `tracks` property of the supplied playlist
+
+## Data Structure
```

Comment:
Data structures should probably be defined before methods; when I'm writing something I don't reference Chapter 7 from Chapter 3 =P